src/java.base/share/classes/jdk/internal/jmod/JmodFile.java

```diff
@@ -52,11 +51,10 @@
     };
 
     public static void checkMagic(Path file) throws IOException {
-        try (InputStream in = Files.newInputStream(file);
-             BufferedInputStream bis = new BufferedInputStream(in)) {
+        try (InputStream in = Files.newInputStream(file)) {
             // validate the header
             byte[] magic = new byte[4];
-            bis.read(magic);
+            in.read(magic);
```

Comment:
While you are there, this can be changed to magic = in.readNBytes(4) and throw IOException if magic.length != 4.

Reply:
Done.